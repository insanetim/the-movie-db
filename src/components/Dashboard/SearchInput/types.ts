export interface SearchInputProps {
  query: string
}

export interface SearchInputHookProps extends SearchInputProps {}

export interface SearchInputHook {
  currentValue: string
  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void
  handleSearch: (value: string) => void
}
