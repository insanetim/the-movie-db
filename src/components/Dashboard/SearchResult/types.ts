import {
  ErrorMsg,
  HandlePagination,
  Query,
} from 'src/interfaces/global.interface'
import { IMoviesList } from 'src/interfaces/movie.interface'

export type SearchResultHookProps = SearchResultProps

export type SearchResultHookReturn = {
  error: ErrorMsg
  handlePagination: HandlePagination
  isLoading: boolean
  movies: IMoviesList | undefined
}

export type SearchResultProps = {
  query: Query
}
