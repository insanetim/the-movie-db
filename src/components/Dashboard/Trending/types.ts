import type { IMoviesList } from 'src/interfaces/movie.interface'

export interface TrendingHook {
  error: null | string
  handlePagination: (page: number) => void
  loading: boolean
  movies: IMoviesList | null
}
