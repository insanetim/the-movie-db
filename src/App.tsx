import { lazy, Suspense } from 'react'
import { Route, Routes } from 'react-router-dom'
import NotificationsRoot from 'src/components/NotificationsRoot'
import ProtectedRoutes from 'src/components/ProtectedRoutes'

export const Login = lazy(() => import('src/pages/Login'))
export const DefaultLayout = lazy(() => import('src/layouts/Default'))
export const Dashboard = lazy(() => import('src/pages/Dashboard'))
export const Lists = lazy(() => import('src/pages/Lists'))
export const Watchlist = lazy(() => import('src/pages/Watchlist'))
export const Favorite = lazy(() => import('src/pages/Favorite'))
export const ListDetails = lazy(() => import('src/pages/ListDetails'))
export const MovieDetails = lazy(() => import('src/pages/MovieDetails'))
export const Cast = lazy(() => import('src/pages/Cast'))
export const PersonDetails = lazy(() => import('src/pages/PersonDetails'))
export const Credits = lazy(() => import('src/pages/Credits'))
export const NotFound = lazy(() => import('src/pages/NotFound'))

const App: React.FC = () => (
  <Suspense fallback={null}>
    <Routes>
      <Route
        element={<Login />}
        path='/login'
      />

      <Route
        element={<DefaultLayout />}
        path='/'
      >
        <Route
          element={<Dashboard />}
          index
        />
        <Route path='movie/:movieSlug'>
          <Route
            element={<MovieDetails />}
            index
          />
          <Route
            element={<Cast />}
            path='cast'
          />
        </Route>
        <Route path='person/:personSlug'>
          <Route
            element={<PersonDetails />}
            index
          />
          <Route
            element={<Credits />}
            path='credits'
          />
        </Route>
        <Route
          element={<NotFound />}
          path='*'
        />

        <Route element={<ProtectedRoutes />}>
          <Route
            element={<Lists />}
            path='lists'
          />
          <Route
            element={<ListDetails />}
            path='list/:listSlug'
          />
          <Route
            element={<Watchlist />}
            path='watchlist'
          />
          <Route
            element={<Favorite />}
            path='favorite'
          />
        </Route>
      </Route>
    </Routes>
    <NotificationsRoot />
  </Suspense>
)

export default App
