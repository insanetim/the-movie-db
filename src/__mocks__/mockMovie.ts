import type { IBackdrop, ICast, ICrew, IMovie, IMovieDetailExtended } from 'src/interfaces/movie.interface'

export const mockMovie: IMovie = {
  adult: false,
  backdrop_path: '/image',
  genre_ids: [1, 2, 3],
  id: 123,
  original_language: 'en',
  original_title: 'test/title',
  overview: 'test/overview',
  popularity: 123,
  poster_path: '/image',
  release_date: '01-01-1999',
  title: 'test/title',
  video: false,
  vote_average: 123,
  vote_count: 123
}

export const mockMovieDetail: IMovieDetailExtended = {
  accountStates: { favorite: false, id: 123, rated: false, watchlist: false },
  adult: false,
  backdrop_path: '/image',
  belongs_to_collection: null,
  budget: 1000000,
  credits: {
    cast: [
      {
        adult: true,
        cast_id: 123,
        character: 'test/character',
        credit_id: '123',
        gender: null,
        id: 123,
        known_for_department: '',
        name: 'test/name',
        order: 123,
        original_name: 'test/name',
        popularity: 123,
        profile_path: '/image'
      }
    ],
    crew: [
      {
        adult: true,
        credit_id: '123',
        department: 'test/department',
        gender: null,
        id: 123,
        job: 'test/job',
        known_for_department: '',
        name: 'test/name',
        original_name: 'test/name',
        popularity: 123,
        profile_path: '/image'
      }
    ],
    id: 123
  },
  genre_ids: [1, 2, 3],
  genres: [{ id: 1, name: 'test/genre' }],
  homepage: null,
  id: 123,
  images: [
    {
      aspect_ratio: 0.5625,
      file_path: '/image',
      height: 900,
      iso_639_1: null,
      vote_average: 123,
      vote_count: 123,
      width: 1600
    }
  ],
  imdb_id: null,
  original_language: 'en',
  original_title: 'test/title',
  overview: 'test/overview',
  popularity: 123,
  poster_path: '/image',
  production_companies: [{ id: 123, logo_path: null, name: 'test/company', origin_country: 'USA' }],
  production_countries: [{ iso_3166_1: 'en', name: 'USA' }],
  release_date: '01-01-1999',
  revenue: 3000000,
  runtime: 150,
  spoken_languages: [{ iso_639_1: 'en', name: 'EN' }],
  status: 'test/status',
  tagline: null,
  title: 'test/title',
  video: false,
  vote_average: 123,
  vote_count: 123
}

export const mockCast: ICast = {
  adult: true,
  cast_id: 123,
  character: 'test/character',
  credit_id: '123',
  gender: null,
  id: 123,
  known_for_department: '',
  name: 'test/name',
  order: 123,
  original_name: 'test/name',
  popularity: 123,
  profile_path: '/image'
}

export const mockCrew: ICrew = {
  adult: true,
  credit_id: '123',
  department: 'test/department',
  gender: null,
  id: 123,
  job: 'test/job',
  known_for_department: '',
  name: 'test/name',
  original_name: 'test/name',
  popularity: 123,
  profile_path: '/image'
}

export const mockImage: IBackdrop = {
  aspect_ratio: 0.5625,
  file_path: '/image',
  height: 900,
  iso_639_1: null,
  vote_average: 123,
  vote_count: 123,
  width: 1600
}
