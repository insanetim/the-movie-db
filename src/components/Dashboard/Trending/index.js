import Trending from './component'

export default Trending
