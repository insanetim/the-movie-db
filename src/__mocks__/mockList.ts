import { IResponse } from 'src/interfaces/global.interface'
import { IList, IListDetails } from 'src/interfaces/list.interface'

import { mockMovie } from './mockMovie'

const mockList: IList = {
  description: 'test/description',
  id: 1234,
  name: 'test/list',
}

const mockListDetails: IListDetails = {
  ...mockList,
  items: [mockMovie],
  page: 1,
  total_pages: 1,
  total_results: 1,
}

const mockListsResponse: IResponse<IList> = {
  page: 1,
  results: [mockList],
  total_pages: 1,
  total_results: 1,
}

export { mockList, mockListDetails, mockListsResponse }
