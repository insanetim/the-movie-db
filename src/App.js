import React from 'react'
import { BrowserRouter, Route, Routes } from 'react-router-dom'

import ProtectedRoutes from './components/ProtectedRoutes'
import Layout from './layout'
import Login from './pages/Login'
import Dashboard from './pages/Dashboard'
import Lists from './pages/Lists'
import Watchlist from './pages/Watchlist'
import Favorites from './pages/Favorites'
import ListDetails from './pages/ListDetails'
import Movie from './pages/Movie'
import './assets/styles/app.scss'

const App = () => (
  <BrowserRouter>
    <Routes>
      <Route element={<ProtectedRoutes />}>
        <Route
          path='/'
          element={<Layout />}
        >
          <Route
            index
            element={<Dashboard />}
          />
          <Route
            path='/lists'
            element={<Lists />}
          />
          <Route
            path='/watchlist'
            element={<Watchlist />}
          />
          <Route
            path='/favorites'
            element={<Favorites />}
          />
          <Route
            path='/list/:listId'
            element={<ListDetails />}
          />
          <Route
            path='/movie/:movieId'
            element={<Movie />}
          />
        </Route>
      </Route>
      <Route
        path='/login'
        element={<Login />}
      />
    </Routes>
  </BrowserRouter>
)

export default App
