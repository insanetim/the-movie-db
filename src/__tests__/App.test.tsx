import { screen } from '@testing-library/react'
import { Location } from 'react-router-dom'
import { mockListDetail } from 'src/__mocks__/mockList'
import { mockMovieDetailExtended } from 'src/__mocks__/mockMovie'
import App, {
  Cast,
  Dashboard,
  DefaultLayout,
  Favorite,
  ListDetails,
  Lists,
  Login,
  MovieDetails,
  NotFound,
  Watchlist,
} from 'src/App'
import { TrendingHookReturn } from 'src/components/Dashboard/Trending/types'
import { ModalRootHookReturn } from 'src/components/ModalRoot/types'
import { NotificationsRootHookReturn } from 'src/components/NotificationsRoot/types'
import { ProtectedRoutesHookReturn } from 'src/components/ProtectedRoutes/types'
import { HeaderHookReturn } from 'src/layouts/Default/Header/types'
import { CastHookReturn } from 'src/pages/Cast/types'
import { DashboardHookReturn } from 'src/pages/Dashboard/types'
import { FavoriteHookReturn } from 'src/pages/Favorite/types'
import { ListDetailsHookReturn } from 'src/pages/ListDetails/types'
import { ListsHookReturn } from 'src/pages/Lists/types'
import { LoginHookReturn } from 'src/pages/Login/types'
import { MovieDetailsHookReturn } from 'src/pages/MovieDetails/types'
import { WatchlistHookReturn } from 'src/pages/Watchlist/types'
import renderWithWrapper from 'src/utils/testHelpers/renderWithWrapper'

const mockedProtectedRoutesHook: ProtectedRoutesHookReturn = {
  isAuthenticated: false,
  location: {} as Location,
}
jest.mock('../components/ProtectedRoutes/hook', () =>
  jest.fn(() => mockedProtectedRoutesHook)
)

const mockedModalRootHook: ModalRootHookReturn = {
  modalProps: null,
  modalType: null,
  onCancel: jest.fn(),
}
jest.mock('../components/ModalRoot/hook', () =>
  jest.fn(() => mockedModalRootHook)
)

const mockedNotificationsRootHook: NotificationsRootHookReturn = {
  hideNotification: jest.fn(),
  notifications: [],
}
jest.mock('../components/NotificationsRoot/hook', () =>
  jest.fn(() => mockedNotificationsRootHook)
)

const mockedHeaderHook: HeaderHookReturn = {
  account: null,
  handleLogOut: jest.fn(),
}
jest.mock('../layouts/Default/Header/hook', () =>
  jest.fn(() => mockedHeaderHook)
)

const mockedLoginHook: LoginHookReturn = {
  handleLogIn: jest.fn(),
  isSubmitting: false,
}
jest.mock('../pages/Login/hook', () => jest.fn(() => mockedLoginHook))

const mockedDashboardHook: DashboardHookReturn = {
  query: '',
}
jest.mock('../pages/Dashboard/hook', () => jest.fn(() => mockedDashboardHook))

const mockedTrendingHook: TrendingHookReturn = {
  error: null,
  handlePagination: () => jest.fn(),
  loading: false,
  movies: null,
}
jest.mock('../components/Dashboard/Trending/hook', () =>
  jest.fn(() => mockedTrendingHook)
)

const mockedListsHook: ListsHookReturn = {
  error: null,
  handleCreateList: jest.fn(),
  handlePagination: jest.fn(),
  lists: null,
  loading: false,
}
jest.mock('../pages/Lists/hook', () => jest.fn(() => mockedListsHook))

const mockedWatchlistHook: WatchlistHookReturn = {
  error: null,
  handleMovieDelete: jest.fn(),
  handlePagination: jest.fn(),
  loading: false,
  movies: null,
}
jest.mock('../pages/Watchlist/hook', () => jest.fn(() => mockedWatchlistHook))

const mockedFavoriteHook: FavoriteHookReturn = {
  error: null,
  handleMovieDelete: jest.fn(),
  handlePagination: jest.fn(),
  loading: false,
  movies: null,
}
jest.mock('../pages/Favorite/hook', () => jest.fn(() => mockedFavoriteHook))

const mockedListDetailsHook: ListDetailsHookReturn = {
  error: null,
  handleListDelete: jest.fn(),
  handleMovieDelete: jest.fn(),
  handlePagination: jest.fn(),
  list: mockListDetail,
  loading: false,
}
jest.mock('../pages/ListDetails/hook', () =>
  jest.fn(() => mockedListDetailsHook)
)

const mockedMovieDetailsHook: MovieDetailsHookReturn = {
  error: null,
  handleFavoriteClick: jest.fn(),
  handleGoToCast: jest.fn(),
  handleWatchlistClick: jest.fn(),
  loading: false,
  movie: mockMovieDetailExtended,
  popoverOpen: false,
  setPopoverOpen: jest.fn(),
}
jest.mock('../pages/MovieDetails/hook', () =>
  jest.fn(() => mockedMovieDetailsHook)
)

const mockedCastHook: CastHookReturn = {
  error: null,
  loading: false,
  movie: mockMovieDetailExtended,
  movieSlug: 'test-movie',
}
jest.mock('../pages/Cast/hook', () => jest.fn(() => mockedCastHook))

describe('App component', () => {
  it('should match snapshot', () => {
    const { asFragment } = renderWithWrapper(<App />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render DefaultLayout', async () => {
    renderWithWrapper(<DefaultLayout />)

    expect(
      await screen.findByText(/insanetim/i, undefined, { timeout: 3000 })
    ).toBeInTheDocument()
  })

  it('should render Login', async () => {
    renderWithWrapper(<Login />)

    expect(await screen.findByText(/log in/i)).toBeInTheDocument()
  })

  it('should render Dashboard', async () => {
    renderWithWrapper(<Dashboard />)

    expect(
      await screen.findByPlaceholderText(/enter/i, undefined, { timeout: 3000 })
    ).toBeInTheDocument()
  })

  it('should render Lists', async () => {
    renderWithWrapper(<Lists />)

    expect(await screen.findByText(/my lists/i)).toBeInTheDocument()
  })

  it('should render Watchlist', async () => {
    renderWithWrapper(<Watchlist />)

    expect(await screen.findByText(/watchlist/i)).toBeInTheDocument()
  })

  it('should render Favorite', async () => {
    renderWithWrapper(<Favorite />)

    expect(await screen.findByText(/favorite/i)).toBeInTheDocument()
  })

  it('should render ListDetails', async () => {
    renderWithWrapper(<ListDetails />)

    expect(await screen.findByText(/title/i)).toBeInTheDocument()
  })

  it('should render MovieDetails', async () => {
    renderWithWrapper(<MovieDetails />)

    expect(
      await screen.findByText(/title/i, undefined, { timeout: 3000 })
    ).toBeInTheDocument()
  })

  it('should render Cast', async () => {
    renderWithWrapper(<Cast />)

    expect(
      await screen.findByText(/back to movie/i, undefined, { timeout: 3000 })
    ).toBeInTheDocument()
  })

  it('should render NotFound', async () => {
    renderWithWrapper(<NotFound />)

    expect(await screen.findByText(/oops!/i)).toBeInTheDocument()
  })
})
