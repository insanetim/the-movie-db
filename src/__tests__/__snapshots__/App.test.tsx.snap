// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`App component should match snapshot 1`] = `<DocumentFragment />`;
